src/nft/components/details/detailsV2/TabbedComponent.tsx

```diff
@@ -0,0 +1,62 @@
+import Row from 'components/Row'
+import { useState } from 'react'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+import { containerStyles } from './shared'
+
+const TabbedComponentContainer = styled.div`
+  height: 288px;
+
+  ${containerStyles}
+`
+
+const TabsRow = styled(Row)`
+  gap: 32px;
+  margin-bottom: 12px;
+  width: 100;
+`
+
+const Tab = styled(ThemedText.SubHeader)<{ isActive: boolean; numTabs: number }>`
+  color: ${({ theme, isActive }) => (isActive ? theme.textPrimary : theme.textTertiary)};
+  line-height: 24px;
+  cursor: ${({ numTabs }) => (numTabs > 1 ? 'pointer' : 'default')};
+
+  &:hover {
+    opacity: ${({ numTabs, theme }) => numTabs > 1 && theme.opacity.hover}};
+  }
+`
+
+export interface Tab {
+  title: React.ReactNode
+  key: string
+  content: JSX.Element
+}
+
+interface TabbedComponentProps {
+  tabs: Tab[]
+  defaultTab: Tab
+  style?: React.CSSProperties
+}
+
+export const TabbedComponent = ({ tabs, defaultTab, style }: TabbedComponentProps) => {
+  const [activeTab, setActiveTab] = useState(defaultTab.key)
+  const activeContent = tabs.find((tab) => tab.key === activeTab)?.content
```

Comment:
array index also makes this easier - you just do tabs[activeTab].content

Reply:
There would also be tradeoffs, later on when we map the tabs to React components we'd also have to track index for knowing which tab is currently active.

Reply:
Why not use `object` or `map` instead? That way, you can still access by object notation. For iteration, you can always use `Object.values`

Reply:
Converted to map